index.js

```diff
@@ -96,8 +96,7 @@ FibonacciHeap.prototype.extractMinimum = function () {
 
     // Merge the children of the minimum node with the root list
     this.minNode = mergeLists(nextInRootList, extractedMin.child, this.compare);
-    if (nextInRootList) {
-      this.minNode = nextInRootList;
+    if (this.minNode) {
```

Comment:
Oh I think I get it, so extracting min from this tree:

```
	___3
   / / |
  _8 5 4
 / |
 9 12
 |
10
```

Would result in this where 8 is the min:

```
  _8---5---4
 / |
 9 12
 |
10
```

But we want this where 4 is the min:

```
  _8---4
 / |   |
 9 12  5
 |
10
```

Is that right?